From c3407ab9d36d94c4d5f8657eceb3e72820cf69dd Mon Sep 17 00:00:00 2001
From: xiaobing tu <xiaobing.tu@intel.com>
Date: Sun, 9 Dec 2012 19:19:23 +0100
Subject: [PATCH 275/276] deadline: Allow 0ms deadline latency, increase the
 read speed

Change a timer compare from after to after-equals, thus allowing
0 timeout and making deadline schedule FIFO.

Signed-off-by: xiaobing tu <xiaobing.tu@intel.com>
Signed-off-by: Jens Axboe <axboe@kernel.dk>
Signed-off-by: flar2 <asegaert@gmail.com>
Signed-off-by: Simarpreet Singh <simar@linux.com>
---
 block/deadline-iosched.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/block/deadline-iosched.c b/block/deadline-iosched.c
index 7bf12d7..08636b0 100644
--- a/block/deadline-iosched.c
+++ b/block/deadline-iosched.c
@@ -230,7 +230,7 @@ static inline int deadline_check_fifo(struct deadline_data *dd, int ddir)
 	/*
 	 * rq is expired!
 	 */
-	if (time_after(jiffies, rq_fifo_time(rq)))
+	if (time_after_eq(jiffies, rq_fifo_time(rq)))
 		return 1;
 
 	return 0;
-- 
1.7.9.5

